main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -217,7 +217,7 @@ class Flix {
       case tast =>
         val ast = PropertyGen.collectProperties(tast)
         val sast = Simplifier.simplify(ast)
-        val lifted = LambdaLift.lift(sast)
+        val lifted = Tailrec.tailrec(LambdaLift.lift(sast))
```

Comment:
I think this is already being addressed in some other branch (typer2?). If not, it is definitely on "the list". A related question is the structure of the later ASTs, because SimplifiedAst is now used by many different phases, both as input/output.